src/utils/DragAndDrop.js

```diff
@@ -181,12 +181,29 @@ define(function (require, exports, module) {
             var errorList = [];
 
             function prepareDropPaths(fileList) {
-                // Convert FileList object to an Array with all *.html files at the end,
-                // since we need to write any .css, .js, etc. resources first such that
-                // Blob URLs can be generated for these resources prior to rewriting an HTML file.
+                // Convert FileList object to an Array with all image files first, then CSS
+                // followed by HTML files at the end, since we need to write any .css, .js, etc.
+                // resources first such that Blob URLs can be generated for these resources
+                // prior to rewriting an HTML file.
+                function rateFileByType(filename) {
+                    var ext = Path.extname(filename);
+
+                    // We want to end up with: [images, ..., js, ..., css, html]
+                    // since CSS can include images, and HTML can include CSS or JS.
+                    // We also treat .md like an HTML file, since we render them.
+                    if(Content.isHTML(ext) || Content.isMarkdown(ext)) {
+                        return 10;
+                    } else if(Content.isCSS(ext)) {
```

Comment:
We do no-rewriting of URLs in JavaScript files, so they can come anywhere.  Same for images, JSON files, etc.  We rewrite CSS files to include images, so the images have to be written first, or won't exist.  We rewrite HTML to fix CSS, JS, and images, so they have to come last.

I think what I have is right.